spec/features/course/assessment/question/duplication_spec.rb

```diff
@@ -6,42 +6,105 @@
 
-        expect(destination_assessment.questions.count).to be(1)
+          duplicated_question = first_destination_assessment.questions.last
+          expect(duplicated_question.title).to eq(question.title)
+          expect(duplicated_question.question_assessments.first.skills.first.id).
+            to eq(question.question_assessments.first.skills.first.id)
+        end
+      end
+
+      context 'upon duplicating deprecated programming language question' do
+        let!(:second_destination_assessment) do
+          create(:assessment, :with_text_response_question, course: course)
+        end
+        let!(:language) { programming_question.specific.language }
+
+        # Disable the language before the test, and enable it after the test
+        # Direct SQL is used to avoid the readonly limitations
+        before do
+          ActiveRecord::Base.connection.execute(
+            "UPDATE polyglot_languages SET enabled = false WHERE id = #{language.id}"
+          )
+          programming_question.reload
+        end
 
-        within all('section', text: source_assessment.questions.first.title).first do
-          click_button 'Duplicate'
+        after do
+          ActiveRecord::Base.connection.execute(
+            "UPDATE polyglot_languages SET enabled = true WHERE id = #{language.id}"
+          )
         end
 
-        wait_for_job
+        scenario 'I can also duplicate that question from one assessment to another' do
+          visit course_assessment_path(course, second_source_assessment)
 
-        expect do
-          find('li', text: destination_assessment.title).click
+          expect(second_destination_assessment.questions.count).to be(1)
+
+          within all('section', text: programming_question.title).first do
+            click_button 'Duplicate'
+          end
+
+          find('li', text: second_destination_assessment.title).click
           expect_toastify('Your question has been duplicated.')
-        end.to change { destination_assessment.questions.count }.by(1)
 
-        original_mcq_question = source_assessment.questions.last
-        duplicated_mcq_question = destination_assessment.questions.last
-        expect(duplicated_mcq_question.title).to eq(original_mcq_question.title)
-        expect(duplicated_mcq_question.question_assessments.first.skills.first.id).
-          to eq(original_mcq_question.question_assessments.first.skills.first.id)
+          wait_for_page
+
+          expect(second_destination_assessment.questions.reload.count).to eq(2)
+          duplicated_question = second_destination_assessment.questions.last
+          expect(duplicated_question.title).to eq(programming_question.title)
+          expect(duplicated_question.question_assessments.first.skills.first.id).
+            to eq(programming_question.question_assessments.first.skills.first.id)
+
+          visit course_assessment_path(course, second_destination_assessment)
```

Comment:
Your test here passes when even before the code change for `validate :validate_language_enabled, unless: :duplicating?`.

Meaning even without `, unless: :duplicating?`, this test can pass.

Fix such that test should fail before code change, and passes after code change please.